core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java

```diff
@@ -0,0 +1,160 @@
+package ca.jrvs.apps.grep;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.log4j.BasicConfigurator;
+
+public class JavaGrepImp implements JavaGrep{
+
+  final Logger logger = LoggerFactory.getLogger(JavaGrep.class);
+
+  private String regex;
+  private String rootPath;
+  private String outFile;
+
+  @Override
+  public String getRegex() {
+    return regex;
+  }
+
+  @Override
+  public void setRegex(String regex) {
+    this.regex = regex;
+  }
+
+  @Override
+  public String getRootPath() {
+    return rootPath;
+  }
+
+  @Override
+  public void setRootPath(String rootPath) {
+    this.rootPath = rootPath;
+  }
+
+  @Override
+  public String getOutFile() {
+    return outFile;
+  }
+
+  @Override
+  public void setOutFile(String outFile) {
+    this.outFile = outFile;
+  }
+
+  @Override
+  public void process() throws IOException {
+    try {
+      List<String> matchedLines = new ArrayList<>();
+
+      for (File file : listFiles(getRootPath())) {
+        for (String line : readLines(file)) {
+          if (containsPattern(line)) {
+            matchedLines.add(line);
+          }
+        }
+      }
+      writeToFile(matchedLines);
+    } catch (IOException ex) {
+      logger.error("USAGE: JavaGrep regex rootPath outFile");

```

Comment:
please pass ex to the logger. otherwise, you will lose the stack track
logger.error(message, ex)